mkgu_packaging/dicarlo/marques/marques_stim_common.py

```diff
@@ -0,0 +1,243 @@
+
+    def update_frame(self):
+        x = (np.arange(self.size_px[1]) - self.size_px[1]/2) / self.px_to_deg - self.pos[1]
+        y = (np.arange(self.size_px[0]) - self.size_px[0]/2) / self.px_to_deg - self.pos[0]
+
+        # all possible degree coordinates in matrices of points
+        self.xv, self.yv = np.meshgrid(x, y)
+
+    def update_tex(self):
+        # make the grating pattern
+        self.tex = (np.sin((self.xv * np.cos(self.orientation) + self.yv * np.sin(self.orientation)) *
+                           self.sf * 2 * np.pi + self.phase) * self.contrast * self.envelope)
+
+    def update_stimulus(self):
+        self.stimulus[self.mask] = np.uint8(((self.tex[self.mask]+1)/2)*self.white)
+        self.stimulus[np.logical_not(self.mask)] = self.gray
+
+    def set_circ_mask(self, rad):
+        # apply operation to put a 1 for all points inclusively within the degree radius and a 0 outside it
+        self.mask = self.xv**2 + self.yv**2 <= rad ** 2
+
+    # same as circular mask but for an annulus
+    def set_annular_mask(self, inner_rad, outer_rad):
+        self.mask = (self.xv ** 2 + self.yv ** 2 <= outer_rad ** 2) * \
+                    (self.xv ** 2 + self.yv ** 2 > inner_rad ** 2)
+
+    def set_gaussian_envelope(self, sig):
+        d = np.sqrt(self.xv**2 + self.yv**2)
+        self.envelope = np.exp(-d**2/(2 * sig**2))
+        self.update_tex()
+
+    def show_stimulus(self):
+        # pyplot stuff
+        self.update_stimulus()
+        my_dpi = 192
+        fig = plt.figure()
+        fig.set_size_inches(self.size_px[1] / my_dpi, self.size_px[0] / my_dpi, forward=False)
+        ax = plt.axes([0, 0, 1, 1])
+        ax.set_axis_off()
+        fig.add_axes(ax)
+        ax.imshow(self.stimulus, cmap='gray')
+        plt.show()
+
+    def save_stimulus(self):
+        # save to correct (previously specified) directory
+        self.update_stimulus()
+        file_name = self.type_name + '_' + self.format_id.format(self.stim_id) + '.png'
+        imageio.imwrite(self.save_dir + os.sep + file_name, self.stimulus)
+        return file_name
+
+
+def load_stim_info(stim_name, data_dir):
+    stim = pd.read_csv(os.path.join(data_dir, 'stimulus_set'), dtype={'image_id': str})
+    image_paths = dict((key, value) for (key, value) in zip(stim['image_id'].values,
+                                                            [os.path.join(data_dir, image_name) for image_name
+                                                             in stim['image_file_name'].values]))
+    stim_set = StimulusSet(stim[stim.columns[:-1]])
+    stim_set.image_paths = image_paths
+    stim_set.identifier = stim_name
+
+    return stim_set
+
+
+def gen_blank_stim(degrees, size_px, save_dir):
+    if not (os.path.isdir(save_dir)):
+        os.mkdir(save_dir)
+
+    stim = Stimulus(size_px=[size_px, size_px], type_name='blank_stim', save_dir=save_dir, stim_id=0)
+
+    stimuli = pd.DataFrame({'image_id': str(0), 'degrees': [degrees]})
```

Comment:
I'm confused by the `'degrees'` here -- at this point it's just pixels no?

Reply:
Here it is actually redundant but see below.

Reply:
should probably delete it from here then, if there's no connection to degrees at this point